drivers/media/i2c/imx219.c

```diff
@@ -767,7 +784,14 @@ static int imx219_enum_mbus_code(struct v4l2_subdev *sd,
 	if (code->index >= (ARRAY_SIZE(codes) / 4))
 		return -EINVAL;
 
-	code->code = imx219_get_format_code(imx219, codes[code->index * 4]);
+	if (code->pad > METADATA_PAD)
+		return -EINVAL;
+
+	if (code->pad == IMAGE_PAD)
+		code->code = imx219_get_format_code(imx219,
+						    codes[code->index * 4]);
+	else
+		code->code = MEDIA_BUS_FMT_SENSOR_DATA;
```

Comment:
You'll return two formats of type MEDIA_BUS_FMT_SENSOR_DATA on the metadata node as this is written.
The ```if (code->index >= (ARRAY_SIZE(codes) / 4))``` is range limiting code->index for the image pad. The metadata pad wants to limit as ```code->index > 0``` as it only has one format.

Reply:
Oops, that test (and the one below) actually got lost when I rebased.  Will fix.

Reply:
Done